client.go

```diff
@@ -29,14 +28,325 @@ func main() {
+func sqlExample(client *hazelcast.Client) {
+	sqlService := client.SQL()
+	// Separate examples that shows SQL usage of Hazelcast Go Client
+	createMappingForCapitals(sqlService)
+	clearCapitals(sqlService)
+	populateCapitals(sqlService)
+	selectAllCapitals(sqlService)
+	selectCapitalNames(sqlService)
+}
+
+func createMappingForCapitals(sqlService sql.Service) {
+	// See: https://docs.hazelcast.com/hazelcast/5.1/sql/mapping-to-maps
+	fmt.Println("Creating a mapping...")
+	mappingQuery := fmt.Sprintf(`
+        CREATE OR REPLACE MAPPING capitals TYPE IMap
+		OPTIONS (
+			'keyFormat' = 'varchar',
+			'valueFormat' = 'varchar'
+		)
+	`)
+	ignored, err := sqlService.Execute(context.Background(), mappingQuery)
+	if err != nil {
+		panic(err)
+	}
+	defer ignored.Close()
+	fmt.Println("The mapping has been created successfully.")
+	fmt.Println("--------------------")
+}
+
+func clearCapitals(sqlService sql.Service) {
+	fmt.Println("Deleting data via SQL...")
+	ignored, err := sqlService.Execute(context.Background(), "DELETE FROM capitals")
+	if err != nil {
+		panic(err)
+	}
+	defer ignored.Close()
+	fmt.Println("The data has been deleted successfully.")
+	fmt.Println("--------------------")
+}
+
+func populateCapitals(sqlService sql.Service) {
+	fmt.Println("Inserting data via SQL...")
+	insertQuery := fmt.Sprintf(`
+        INSERT INTO capitals VALUES
+			('Australia','Canberra'),
+            ('Croatia','Zagreb'),
+			('Czech Republic','Prague'),
+			('England','London'),
+			('Turkey','Ankara'),
+			('United States','Washington, DC');
+	`)
+	ignored, err := sqlService.Execute(context.Background(), insertQuery)
+	if err != nil {
+		panic(err)
+	}
+	defer ignored.Close()
+	fmt.Println("The data has been inserted successfully.")
+	fmt.Println("--------------------")
+}
+
+func selectAllCapitals(sqlService sql.Service) {
+	fmt.Println("Retrieving all the data via SQL...")
+	ctx := context.Background()
+	result, err := sqlService.Execute(ctx, "SELECT * FROM capitals")
+	if err != nil {
+		fmt.Errorf("Querying: %w", err)
+	}
+	iterator, err := result.Iterator()
+	if err != nil {
+		fmt.Errorf("Acquaring iterator: %w", err)
+	}
+	for iterator.HasNext() {
+		row, err := iterator.Next()
+		if err != nil {
+			panic(err)
+		}
+		country, err := row.Get(0)
+		city, err := row.Get(1)
+		fmt.Printf("%s - %s\n", country, city)
+	}
+	fmt.Println("--------------------")
+}
+
+func selectCapitalNames(sqlService sql.Service) {
+	ctx := context.Background()
+	fmt.Println("Retrieving the capital name via SQL...")
+	result, err := sqlService.Execute(ctx, "SELECT __key, this FROM capitals WHERE __key = ?", "United States")
+	iter, err := result.Iterator()
+	if err != nil {
+		fmt.Errorf("acquaring iterator: %w", err)
+	}
+	for iter.HasNext() {
+		row, err := iter.Next()
+		if err == nil {
+			fmt.Errorf("iterating: %w", err)
+		}
+		country, err := row.GetByColumnName("__key")
+		city, err := row.GetByColumnName("this")
+		fmt.Printf("%s - %s\n", country, city)
```

Comment:
```suggestion
		fmt.Printf("Country name: %s; Capital name: %s\n", country, city)
```
I guess the print message is a bit different on the java sample